ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/math/Greatest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.math;
+
+import com.google.common.collect.Streams;
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.function.udf.UdfSchemaProvider;
+import io.confluent.ksql.schema.ksql.SqlArgument;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.DecimalUtil;
+import io.confluent.ksql.util.KsqlConstants;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+@UdfDescription(
+    name = "greatest",
+    category = FunctionCategory.MATHEMATICAL,
+    description = "Returns the highest value among a variable number of consistently typed columns",
+    author = KsqlConstants.CONFLUENT_AUTHOR
+)
+public class Greatest {
+
+  @Udf
+  public Double greatest(@UdfParameter final Double val, @UdfParameter final Double... vals) {
+
+    final Stream<Double> toConcat = Stream.of(val);
+
+    return Streams.concat(toConcat, Arrays.stream(vals))
+        .filter(Objects::nonNull)
```

Comment:
On handling nulls: I chose to return null only if every argument is null, otherwise returning the highest (or lowest in the case of `least`) value. This is in line with how Postgres implements these functions, but this isn't consistent (for example MySQL returns null if any entry is null). I'd like input on what you think is best here @colinhicks

Reply:
Your choice sounds good to me 👍